lib/builder/context.ex

```diff
@@ -0,0 +1,17 @@
+defmodule Burrito.Builder.Context do
+  use TypedStruct
+
+  alias Burrito.Builder.Target
+
+  @type erts_location :: nil | {:release | :local | :url | :unpacked, term()}
```

Comment:
When will this be `nil`? I might be missing something, but I believe it's always set.

It may also be worth splitting these cases out of the pair so that they can be given more specific types: the `:release` case in particular doesn't seem to use the 2nd element of the tuple:

```elixir
  @type erts_location ::
          :release
          | {:local, Path.t()}
          | {:unpacked, Path.t()}
          | {:url, URI.t()}
```

Note that I changed the `:url` case to use `URI.t()`. `Req` supports accepting `URI.t()`, but if you use this type you'll want to call `URI.parse/1` in order to construct it, so that we aren't forcing Elixir 1.13 as a requirement.